src/OnlineSales/Controllers/OrderItemsController.cs

```diff
@@ -93,5 +93,31 @@ public override async Task<ActionResult> Delete(int id)
 
             return NoContent();
         }
+
+        protected override List<OrderItem> GetMappedRecords(List<OrderItemImportDto> records)
+        {
+            foreach (var item in records)
+            {
+                if (item.OrderId > 0)
+                {
+                    continue;
+                }
+
+                if (item.OrderId == 0 && !string.IsNullOrEmpty(item.OrderRefNo))
+                {
+                    var referencedOrder = dbContext!.Orders!.Where(o => o.RefNo.ToLower() == item.OrderRefNo.ToLower()).FirstOrDefault();
```

Comment:
It will definitely generate a separate select for every imported record - very heavy and unnecessary load on the database. Please find a way to make predictable amount of select queries to the database. It must be 1, 2, 3 maximum - i.e. it should be the same number of queries independently of the amount of imported records.